src/DynamicDiscreteSamplers.jl

```diff
@@ -220,10 +220,10 @@ end
 
 function _getindex(m::Memory{UInt64}, i::Int)
     @boundscheck 1 <= i <= m[1] || throw(BoundsError(_FixedSizeWeights(m), i))
-    j = 2i + 10490
-    pos = m[j]
+    j = i + 10491
+    exponent = m[j] & 2047
+    pos = (m[j] - exponent) >> 11
```

Comment:
```suggestion
    pos = m[j] >> 11
```
Won't the `>> 11` shift all those exponent bits off the edge?

Reply:
right!